TemplateApplication/Onboarding/NotificationPermissions.swift

```diff
@@ -41,30 +42,37 @@ struct NotificationPermissions: View {
                     action: {
                         do {
                             notificationProcessing = true
-                            // HealthKit is not available in the preview simulator.
-                            if ProcessInfo.processInfo.environment["XCODE_RUNNING_FOR_PREVIEWS"] == "1" {
+                            // Notification Authorization is not available in the preview simulator.
+                            if ProcessInfo.processInfo.isPreviewSimulator {
                                 try await _Concurrency.Task.sleep(for: .seconds(5))
                             } else {
                                 try await scheduler.requestLocalNotificationAuthorization()
                             }
                         } catch {
                             print("Could not request notification permissions.")
                         }
-                        completedOnboardingFlow = true
```

Comment:
I like that this is now done automatically, this will create nice diffs if we append anything to the onboarding flow. I remember that this was a pain point in the @CS342 class last year.